Fix bug during initial download of dependencies - issue #4407

AndreasArendt opened the pull request:
rework download and unzip dependencies function duing initial bootstrapping

Tomcus reviewed (changes requested):
Hi. Thanks for upgrading our bootstrap to `requests`.

One small nitpick: Your commit message title is overflowing, you might want to consider shortening it and put more information into the commit body.

AndreasArendt commented on requirements.txt:
```diff
@@ -15,3 +15,4 @@ python-bidi==0.4.2
 pyyaml~=6.0
 qoi~=0.5.0
 setuptools~=70.1.0
+requests==2.32.3
```
Thanks for letting me know. Will fix this today in the evening.

AndreasArendt replied:
@Tomcus, do you mean the utils/holly/build-requirements.txt or which one do you have in mind?

Tomcus replied:
Yes

AndreasArendt replied:
done

Tomcus commented:
The docker container still doesn't build. You'll need to change it too. The requirements are installed by the bootstrap.py, so you can't call bootstrap.py with `requests`. We need to install it before running it or include it conditionaly later after the requirements are installed.

AndreasArendt commented:
will have a look, thank you for letting me know!

Tomcus commented on lib/Prusa-Firmware-MMU/utils/holly/Dockerfile:
This is just a subrepo of the MMU firmware, you shouldn't change the files here. Please remove this change.

AndreasArendt replied:
sorry will do - was not yet done yesterday, but thx!

Tomcus commented on utils/holly/Dockerfile:
```diff
@@ -1,8 +1,9 @@
-    apt-get install libjpeg-dev git python3 python3-pip python3-venv -y
+    apt-get install libjpeg-dev git python3 python3-pip python3-venv python3-requests -y    
 WORKDIR /work
+RUN mkdir .dependencies
```
Do we really need to create the `.dependencies` folder. I think the `bootstrap.py` should do that. But maybe it is broken on windows?

AndreasArendt replied:
it threw an error in my docker otherwise. I thought the docker could take care of, but sure we could try without it.

AndreasArendt replied:
fixed this in bootstrap.py now

Tomcus reviewed (changes requested):
Otherwise good job.

Tomcus commented:
Also formating check failed. You can run `pre-commit run --all-files` to fix that.

AndreasArendt commented:
thank you for the feedback - was not yet finished yesterday and wanted to double check some things.
How about jenkins here? Is there still something missing as it fails?

Tomcus commented:
Just the formating and sadly we are being rate limited (not your fault - we had to quickly redo part of the infrastructure and couldn't do it 1:1), so sometimes download fails, but retry should work. I will retry jobs for you if they fail because of ratelimited download.

AndreasArendt commented:
Okay, then I will fix the formatting and remove the changes for MMU firmware on the docker file (sorry for that).
Will let you know

AndreasArendt commented:
@Tomcus, so did the formatting now.
* It automatically created the `doc/logging_components.md` - hope that's ok
* Changed the `.dependencies` folder generation now and added it to bootstrapping - makes also more sense in my mind
* docker is running, build script and bootstrapping aswell
* reverted the changes in `lib/Prusa-Firmware-MMU/utils/holly/Dockerfile`

let me know if anything is missing

Tomcus commented:
Hi the formating check is still not passing. It looks like that the changes that your pre-commit did to `doc/logging_components.md` are not compatible with the one generated by linux/docker version. I would probably just revert the changes to that file.

AndreasArendt commented:
Thanks for checking. Okay, then this seems to be a windows / linux issue.
I could not commit without adding the markdown file. Will revert it and see how I commit then without `pre-commit` hook complaining.

Tomcus commented:
If you run `git commit --no-verify` you can skip the pre commit hooks.

AndreasArendt commented:
Thanks for the tip! Did already revert the markdown.
Anything else missing?

Tomcus commented:
I still see it. Did you push the changes? :)

AndreasArendt commented:
so sorry, not on my usual machine so messed it up

Tomcus commented:
It looks ok, but the pipeline fails with the ratelimiting problem. I will try to rerun the pipeline untill it passes. Otherwise I am ok with the changes. Can you please squash the commits into one. I will then aprove it and merge it.

AndreasArendt commented:
done. Thank you very much for giving me all these tips! It's my first contribution and I will remember them next time. Appreciate!

Tomcus reviewed (approved):
Please don't merge it
